Add a changelog section, common to both variants

nord-erik opened the pull request:
Of course there are some differences between BigQuery and Google Cloud in terms 
of documentation. But I decided for simplicity to have the same changelog for
both -- it's the same source code for both. Interpreted both git history and code 
changes to figure out major differences between the different releases.

[Trello Card](https://trello.com/c/Byf10wFf/1203-dataflow-flex-template-release-notes)

nord-erik committed:
Add a changelog section, common to both variants

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

nord-erik committed:
Update intro section to changelog

nord-erik committed:
Update changelog with all relevant releases to neo4j

extracted using command:

```
git log --format="%H" --since="2024-01-09 00:00:00" -- src/ \
  | xargs -I{} bash -c 'printf "%s %s\n" "{}" "$(git tag --contains {} | head -n1)"'\
  | awk '{print $2;}'\
  | uniq\
  | awk '{print "==" $1}'
```

nord-erik committed:
Add logs since 2024-01-01

Changes were determined if they shipped in a release with:

```
git log --format="%H" --since="2024-01-10 00:00:00" -- src \
  | xargs -I{} bash -c 'printf "%s %s\n" "{}" "$(git tag --contains {} | head -n1)"'
```

Then to determine what actually did change:

```
git log --format="%s" 2025-01-22-01_RC01..2025-01-26-00_RC00 -- src \
  | awk '{print "* "$0;}'
```

Did not include test changes, minor changes etc.
Reworded to make sense from someone reading just this changelog

nord-erik committed:
Update releases to have release prefix

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

nord-erik committed:
Fix spelling errors

stefano-ottolenghi committed:
Merge branch 'dev' into en/changelog

stefano-ottolenghi commented:
@nord-erik :wave: #29 fixed the previews not being available. You will now find that the top link to surge.sh works, with a landing that gives you access to the two docsets. However, the `Changelog` links don't work. That's because the related pages are not included in `content-nav.yml`, and the docs automation strips away from publishing anything that's not explicitly referenced in the navigation. Adding an entry to the navigation for the changelog (in both docsets) should fix the issue.

fbiville committed:
docs: add change log to navigation

fbiville committed:
docs: add instructions to check release status

nord-erik commented:
Hi @stefano-ottolenghi , thanks a lot for the info! I see Florent is already on it

fbiville committed:
docs: cover all history, trim down documented releases

nord-erik committed:
fix build warning

nord-erik committed:
add changelog to google cloud content-nav

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+
+== Release 2025-01-26-00_RC00
+
+* Add support for calendar-based version names of the Neo4j database
```
What does this mean? What are some examples of `version names` I would see?

nord-erik replied:
This refers to this change specifically:
https://github.com/GoogleCloudPlatform/DataflowTemplates/commit/e6a515b15a39d01da548c34c94e57d2db936a8e8

I.e. a small change to make sure the new version format of neo4j since version 2025 will still work.

Is this unnecessary information to highlight or should we re-formulate it a bit?

stefano-ottolenghi replied:
Ah, right, I see. I don't know if it's unnecessary information to highlight - if we do highlight it, I would phrase it as "Support for self-managed deployments of Neo4j 2025.x", because that's what's relevant to customers.

nord-erik replied:
Thank you so much for the input, I will add this!

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+
+Releases are gradually deployed, some regions may be updated before others.
+To check the release status of https://cloud.google.com/compute/docs/regions-zones[a particular region], you can run `gsutil ls gs://dataflow-templates-$\{region}/` (e.g. `gsutil ls gs://dataflow-templates-europe-west2/`).
+
```
Shall we also mention that it is not possible to run any version other than _latest_?

nord-erik replied:
Yes, let me edit this in

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+== Release 2025-01-22-01_RC00
+
+* Switch to Java 17
+* Updated Neo4j driver version from v4.4.18 to v5.27.0
```
Are we not yet running on 5.28.8 (i.e. the LTS)?

nord-erik replied:
It's a retroactive change-log, this date 2025-01-22, the LTS version 5.28.x did not exist yet.

We are still on v5.27.0, great observation -- maybe we should upgrade this to a later version, will hear with team

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+== Release 2025-01-07-00_RC01
+
+* Remove node target's existence constraint
+* Fix an issue with custom Cypher queries
```
I find these sort of entries quite useless from a user's perspective. Is it a very-hard-to-describe issue, or can we at least hint at what it affected?

nord-erik replied:
Did some git history research, read some code. Posting an update to what changed.

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+
+== Release 2024-12-10-00_RC00
+
+* Added support for BigQuery properties 'query_temp_project' and 'query_temp_dataset'
```
```suggestion
* Added support for BigQuery properties `query_temp_project` and `query_temp_dataset`
```

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,101 @@
+
+== Release 2024-09-18-00_RC00
+
+* Introduce new import specification (https://github.com/neo4j/import-spec)
```
```suggestion
* Introduce new import specification (https://github.com/neo4j/import-spec), on which Dataflow templates base their import specifications.
```

nord-erik committed:
Update common-content-dataflow/modules/ROOT/partials/changelog.adoc

Co-authored-by: Stefano Ottolenghi <stefano.ottolenghi@neo4j.com>

nord-erik committed:
Update common-content-dataflow/modules/ROOT/partials/changelog.adoc

Co-authored-by: Stefano Ottolenghi <stefano.ottolenghi@neo4j.com>

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

nord-erik committed:
update changelog

- added notice that latest version is alawys used
- consistently do not full stop at end of entries
- clarified calendar version support
- clarified custom query issue (stating symptons -- core issue very complex to explain)

stefano-ottolenghi commented on common-content-dataflow/modules/ROOT/partials/changelog.adoc:
```diff
@@ -0,0 +1,102 @@
+
+Releases are gradually deployed, some regions may be updated before others.
+To check the release status of https://cloud.google.com/compute/docs/regions-zones[a particular region], you can run `gsutil ls gs://dataflow-templates-$\{region}/` (e.g. `gsutil ls gs://dataflow-templates-europe-west2/`).
+It is not possible to run any other version than the *latest* version of your region.
```
```suggestion
It is not possible to run any version other than the *latest* one deployed in a given region.
```

stefano-ottolenghi committed:
Update common-content-dataflow/modules/ROOT/partials/changelog.adoc

neo4j-docops-agent commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-dataflow-connector-28.surge.sh

New pages:
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-bigquery/1/changelog)
- [Changelog](https://neo4j-docs-dataflow-connector-28.surge.sh/dataflow-google-cloud/1/changelog)
<!-- Sticky Pull Request Commentdocs-pr-changes -->
